subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/SwiftXCTestIntegrationTest.groovy

```diff
@@ -493,29 +476,137 @@ apply plugin: 'swift-library'
+                public func testGetMessage() {
+                    let util = Util()
+                    util.doIt()
+                    XCTAssert(true)
+                }
+            } 
+        """
         when:
         succeeds 'test'
 
         then:
-        result.assertTasksExecuted(':greeter:compileDebugSwift', ':greeter:compileTestSwift', ':greeter:linkDebug',
-            ':greeter:linkTest', ':greeter:bundleSwiftTest', ':greeter:xcTest', ':greeter:test', ':compileDebugSwift',
-            ':compileTestSwift', ':linkTest', ':bundleSwiftTest', ':xcTest', ':test')
+        result.assertTasksExecuted(
+            ':log:compileDebugSwift', ':log:linkDebug',
+            ':hello:compileDebugSwift', ':hello:linkDebug',
+            ':compileDebugSwift', ':compileTestSwift', ":relocateMainForTest", ':linkTest', bundleOrInstallTask(), ':xcTest', ':test')
     }
 
-    // TODO: Need to support _main symbol duplication
-    @Requires(TestPrecondition.MAC_OS_X)
     def 'can run xctest in swift package manager layout'() {
+        given:
+        def app = new SwiftAppWithLibraries()
+        settingsFile << """
+            rootProject.name = 'app'
+            include 'hello', 'log'
+        """
+        buildFile << """
+            apply plugin: 'swift-executable'
+            executable {
+                source.from rootProject.file('Sources/App')
+            }
+            xctest {
+                source.from rootProject.file('Tests/AppTests')
+            }
+            dependencies {
+                implementation project(':hello')
+            }
+            
+            project(':hello') {
+                apply plugin: 'swift-library'
+                library {
+                    source.from rootProject.file('Sources/Hello')
+                }
+                dependencies {
+                    api project(':log')
+                }
+            }
+            project(':log') {
+                apply plugin: 'swift-library'
+                library {
+                    source.from rootProject.file('Sources/Log')
+                }
+            }
+        """
+
+        app.executable.writeToProject(file('Sources/App'))
+        app.greeter.writeToProject(file('Sources/Hello'))
+        app.logger.writeToProject(file('Sources/Log'))
+
+        file("Sources/App/Util.swift") << """
+            import Hello 
+            
+            public class Util {
+                public init() {}
+                public func doIt() {
+                    let greeter = Greeter()
+                    greeter.sayHello()
+                }
+            }
+        """
+        file('Tests/AppTests/UtilTest.swift') << """
+            import XCTest
+            import App
+            
+            public class UtilTest : XCTestCase {
+                public func testGetMessage() {
+                    let util = Util()
+                    util.doIt()
+                    XCTAssert(true)
+                }
+            } 
+        """
```

Comment:
I don't get why these files are created here. Shouldn't they be part of the `SwiftAppWithLibraries` fixture?

Reply:
SwiftAppWithLibraries doesn't have tests for the projects.  I meant to change this around to use main() once I could do that.  Let me change that.